### PR TITLE
Add mixed precision training example

### PR DESCRIPTION
**Pull request by assassin1991**
<details>
   <summary><b>Add mixed precision training example</b></summary>

Usage:
python -u mnist_trainer.py -v
to override ops with autocast:
LOWER_LIST=<path_to_bf16_ops> python -u mnist_trainer.py -v

- [ ] Was this discussed/agreed via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

## What does this PR do?
Added mixed precision training example

Fixes # (issue).

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in GitHub issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: examples/pytorch/mnist_trainer.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+if __name__ == "__main__":
+    # Get options
+    options = parse_args()
+    if options.verbose:
+        print(f"Running MNIST mixed precision training with options: {options}")
+
+    # Run model and print accuracy
+    for run_type in options.run_types:
+        seed_everything(42)
+        run_model(run_type, options.plugins, options.verbose)
```
Comment (Borda) [inline examples/pytorch/mnist_trainer.py:112]: why not use Lightning CLI?

Comment (assassin1991) [reply examples/pytorch/mnist_trainer.py:112]: tested with LightningCLI, currently it does not seem to support setting accelerator class in trainer config. Setting it with "hpu" picks HPUAccelerator from lightning and not from lightning_habana

**Review by jerome-habana — APPROVED** [review]

File: examples/pytorch/mnist_trainer.py (hunk 1 above)
Comment (Borda) [reply examples/pytorch/mnist_trainer.py:112]: Cc: @awaelchli

Comment (jerome-habana) [reply examples/pytorch/mnist_trainer.py:112]: @Borda @assassin1991 lets create an issue and track it through for CLI and external accelerators

Comment (assassin1991) [reply examples/pytorch/mnist_trainer.py:112]: Issue created: https://github.com/Lightning-AI/lightning-Habana/issues/55

**Pull request merged**